### PR TITLE
106 encryption/decryption

### LINKED ISSUE #106: Encryption/decryption of json

### PR DESCRIPTION
**Pull request by MathiasMeuleman**
Will close #106 

Savegames are now encrypted using a Base64 encryption.
Files are stored as .ses, these are automatically read on loadGameScene startup.

Please review

<feature, review

### CODE REVIEW

File: src/main/java/nl/tudelft/ti2206/group9/gui/GameScene.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
 
 import static nl.tudelft.ti2206.group9.ShaftEscape.OBSERVABLE;
 
+
```
Comment (mdingjan) [inline src/main/java/nl/tudelft/ti2206/group9/gui/GameScene.java]: Revert this change.

<review, layout

File: src/main/java/nl/tudelft/ti2206/group9/util/Parser.java [hunk 2]
```diff
@@ -44,13 +47,24 @@ private Parser() { }
 	 */
 	static void loadGame(final String path) {
```
Comment (mdingjan) [inline src/main/java/nl/tudelft/ti2206/group9/util/Parser.java]: Method is 3 lines too long.

<review,organization

File: src/main/java/nl/tudelft/ti2206/group9/util/Writer.java [hunk 3]
```diff
@@ -6,12 +6,14 @@
...
 import java.io.IOException;
 import java.io.OutputStreamWriter;
+import java.io.UnsupportedEncodingException;
 
 import nl.tudelft.ti2206.group9.level.State;
 import nl.tudelft.ti2206.group9.util.GameObserver.Category;
 import nl.tudelft.ti2206.group9.util.GameObserver.Error;
 
 import org.json.simple.JSONObject;
+import sun.misc.BASE64Encoder; //NOPMD - I need this package
```
Comment (mdingjan) [inline src/main/java/nl/tudelft/ti2206/group9/util/Writer.java]: Add @SuppressWarnings("restriction") for using this import.

<review,organization

**Comment by mdingjan** [comment]
I reviewed your changes. I added suggestions above and processed the obvious remaining obvious SAT violations. Travis' builds succeed now.
Your codes looks good! I really like that we now have our own file extension. :D

This branch is ready to be merged, in my opinion.

<review

File: checkstyle.xml [hunk 4]
```diff
@@ -30,7 +30,6 @@
     <module name="StaticVariableName"/>
     <module name="TypeName"/>
     <module name="AvoidStarImport"/>
-    <module name="IllegalImport"/>
```
Comment (mpsijm) [inline checkstyle.xml]: Why is this rule removed? I see why this is not needed, but I would like to know why, and this should also be removed in `checkstyle_text.xml`.

<revew, configuration

File: src/main/java/nl/tudelft/ti2206/group9/util/Parser.java [hunk 5]
```diff
@@ -10,18 +19,14 @@
 import org.json.simple.parser.JSONParser;
 import org.json.simple.parser.ParseException;
 
-import java.io.BufferedReader;
-import java.io.File;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.net.URL;
+import sun.misc.BASE64Decoder; //NOPMD - I need this package
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/util/Parser.java]: And you're sure there is no other plugin that supports base-64 encoding?

File: src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java [hunk 6]
```diff
@@ -113,7 +113,7 @@ public void test() throws IOException { //NOPMD - assert is done in subs.
 		playerDies();
 		clickPopup(DEATH_TOMAIN);
 
-		mainMenu(MAIN_QUIT);
+		Platform.runLater(stage::close);
```
Comment (mpsijm) [inline src/test/java/nl/tudelft/ti2206/group9/gui/EndToEndTest.java]: This change was really needed to make the E2E run, right? Please confirm, if so, then leave this be.

<review,testing

File: src/main/java/nl/tudelft/ti2206/group9/util/Parser.java (hunk 2 above)
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/util/Parser.java]: Not anymore apparently :)

<review, layout

**Comment by MKerkhof** [comment]
SATs didn't give any warnings, besides that, the functionality looks good and real "hufter-proof". Loading also works oke so I think this feature is ready to be merged!

<review

File: src/main/java/nl/tudelft/ti2206/group9/util/Parser.java (hunk 2 above)
Comment (mdingjan) [inline src/main/java/nl/tudelft/ti2206/group9/util/Parser.java]: I fixed it. ;)

<review

**Comment by mpsijm** [comment]
As I fixed your SAT warnings, I also approve this branch to be merged. v1.3, here we come!

<review
